Microsoft.Toolkit.Uwp.SampleApp/Controls/ExtendedHamburgerMenu.cs

```diff
@@ -0,0 +1,630 @@
+
+        private Button _hamburgerButton;
+        private ListView _buttonsListView;
+        private ListView _optionsListView;
+        private Grid _samplePickerGrid;
+        private GridView _samplePickerGridView;
+        private Border _contentShadow;
+        private Grid _searchGrid;
+        private TextBlock _titleTextBlock;
+        private AutoSuggestBox _searchBox;
+        private Button _searchButton;
+
+        private Canvas _moreInfoCanvas;
+        private FrameworkElement _moreInfoContent;
+        private Image _moreInfoImage;
+
+        /// <summary>
+        /// Event raised when an item is clicked
+        /// </summary>
+        public event ItemClickEventHandler SamplePickerItemClick;
+
+        private Sample _currentSample;
+
+        public string Title
+        {
+            get { return (string)GetValue(TitleProperty); }
+            set { SetValue(TitleProperty, value); }
+        }
+
+        public static readonly DependencyProperty TitleProperty =
+            DependencyProperty.Register("Title", typeof(string), typeof(ExtendedHamburgerMenu), new PropertyMetadata(string.Empty));
+
+        public Sample CurrentSample
+        {
+            get
+            {
+                return _currentSample;
+            }
+
+            set
+            {
+                _currentSample = value;
+                var noop = SetHamburgerMenuSelection();
+            }
+        }
+
+        public void HideSamplePicker()
+        {
+            if (SetupSamplePicker())
+            {
+                _samplePickerGrid.Visibility = Visibility.Collapsed;
+            }
+        }
+
+        public async void ShowSamplePicker(Sample[] samples = null)
+        {
+            if (!SetupSamplePicker())
+            {
+                return;
+            }
+
+            if (samples == null && _currentSample != null)
+            {
+                var category = await Samples.GetCategoryBySample(_currentSample);
+                if (category != null)
+                {
+                    samples = category.Samples;
+                }
+            }
+
+            if (samples == null)
+            {
+                samples = (await Samples.GetCategoriesAsync()).FirstOrDefault()?.Samples;
+            }
+
+            if (samples == null)
+            {
+                return;
+            }
+
+            _samplePickerGridView.ItemsSource = samples;
+
+            if (_currentSample != null && samples.Contains(_currentSample))
+            {
+                _samplePickerGridView.SelectedItem = _currentSample;
+            }
+            else
+            {
+                _samplePickerGridView.SelectedItem = null;
+            }
+
+            _samplePickerGrid.Visibility = Visibility.Visible;
+        }
+
+        public async Task StartSearch(string startingText = null)
+        {
+            if (_searchBox == null || _searchBox.Visibility == Visibility.Visible)
```

Comment:
I think it'd be actually nice if you hit 'enter' in the blank searchbox that it shows all the controls (it does this if you type something and then hit backspace).

Did you get Shane to test these changes on the XBox?

Reply:
Good idea, fixed. Tested it on Xbox too

Reply:
Why not just show everything as soon as you click the search button and then filter down from there

Reply:
Having the animation when filtering is very jarring. This should be disabled